src/compiler/crystal/semantic/type_declaration_processor.cr

```diff
@@ -631,6 +621,40 @@ struct Crystal::TypeDeclarationProcessor
     nil
   end
 
+  private def remove_duplicate_instance_vars_declarations
+    # All the types that we checked for duplicate variables
+    duplicates_checked = Set(Type).new
+    remove_duplicate_instance_vars_declarations(@program, duplicates_checked)
+  end
+
+  private def remove_duplicate_instance_vars_declarations(type : Type, duplicates_checked : Set(Type))
```

Comment:
Out of curiosity, wouldn't it make sense just to use defaults and avoid the other definition?

Reply:
I don't think using defaults makes sense. It's not like I'd like to call this with one of the arguments and not the other. But... if you feel like it, feel free to adapt this PR to your taste :-)

Reply:
No, this is fine, I was just curious. And I just saw a resolved previous comment about this same thing, sorry for the nagging!